events/janus_mqttevh.c

```diff
@@ -1005,3 +1181,52 @@ static void *janus_mqttevh_handler(void *data) {
 	JANUS_LOG(LOG_VERB, "Leaving MQTTEventHandler handler thread\n");
 	return NULL;
 }
+
+int janus_mqttevh_client_get_response_code(MQTTAsync_failureData *response) {
+	return response ? response->code : 0;
+}
+
+#ifdef MQTTVERSION_5
+int janus_mqttevh_client_get_response_code5(MQTTAsync_failureData5 *response) {
+	return response ? response->code : 0;
+}
+
+void janus_mqttevh_add_properties(GArray *user_properties, MQTTProperties *properties) {
+	if(user_properties == NULL || user_properties->len == 0) {
+		return;
+	}
+
+	for(uint i = 0; i < user_properties->len; i++) {
+		MQTTProperty *property = &g_array_index(user_properties, MQTTProperty, i);
+		int rc = MQTTProperties_add(properties, property);
+		if(rc != 0) {
+			JANUS_LOG(LOG_ERR, "Failed to user properties to MQTT response\n");
+		}
+	}
+}
+
+void janus_mqttevh_set_add_user_property(gpointer item_ptr, gpointer user_data_ptr) {
+	janus_config_item *item = (janus_config_item*)item_ptr;
+	if(item->value != NULL) {
+		return;
+	}
+
+	janus_mqttevh_set_add_user_property_user_data *user_data = (janus_mqttevh_set_add_user_property_user_data*)user_data_ptr;
+	GList *key_value = janus_config_get_items(user_data->config, item);
+	if(key_value == NULL || g_list_length(key_value) != 2) {
+		JANUS_LOG(LOG_ERR, "Expected a key-value pair\n");
+		return;
+	}
+
+	janus_config_item *key_item = (janus_config_item*)g_list_first(key_value)->data;
+	janus_config_item *value_item = (janus_config_item*)g_list_last(key_value)->data;
+
+	MQTTProperty property;
+	property.identifier = MQTTPROPERTY_CODE_USER_PROPERTY;
+	property.value.data.data = g_strdup(key_item->value);
+	property.value.data.len = strlen(key_item->value);
+	property.value.value.data = g_strdup(value_item->value);
+	property.value.value.len = strlen(value_item->value);
```

Comment:
These seem to be giving for granted that both `key_item` and `value_item` will have a valid `value`, which may not be the case. If `value` is `NULL` I think `g_strdup` may crash, so it might be better to make sure `value` exists.